xarray/core/treenode.py

```diff
@@ -395,15 +396,18 @@ def subtree(self: Tree) -> Iterator[Tree]:
         """
         An iterator over all nodes in this tree, including both self and all descendants.
 
-        Iterates depth-first.
+        Iterates breadth-first.
 
         See Also
         --------
         DataTree.descendants
         """
-        from xarray.core.iterators import LevelOrderIter
-
-        return LevelOrderIter(self)
+        # https://en.wikipedia.org/wiki/Breadth-first_search#Pseudocode
+        queue = collections.deque([self])
+        while queue:
+            node = queue.popleft()
+            yield node
+            queue.extend(node.children.values())
```

Comment:
Replacing the entire `iterators.py` file with 6 lines is so clever it's almost rude 🤣